sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala

```diff
@@ -306,10 +305,10 @@ class ParquetFileFormat
     hadoopConf.set(ParquetInputFormat.READ_SUPPORT_CLASS, classOf[ParquetReadSupport].getName)
     hadoopConf.set(
       ParquetReadSupport.SPARK_ROW_REQUESTED_SCHEMA,
-      ParquetSchemaConverter.checkFieldNames(requiredSchema).json)
+      requiredSchema.json)
     hadoopConf.set(
       ParquetWriteSupport.SPARK_ROW_SCHEMA,
-      ParquetSchemaConverter.checkFieldNames(requiredSchema).json)
```

Comment:
I don't think we need to check field names at this last step, cc @dongjoon-hyun

Reply:
Thank you for pinging me. I see. That logic is introduced at [SPARK-13883](https://github.com/cloud-fan/spark/blame/da54abfd8730ef752eca921089bcf568773bd24a/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetRelation.scala#L302) about 2 years ago by @marmbrus .

Reply:
This is for the read path. Not sure whether this also matters. 

The other checks are for DDL.

Reply:
at read path, the parquet files are already there, so the field names must be valid :)